javaslang/src/test/java/javaslang/collection/VectorPropertyTest.java

```diff
@@ -0,0 +1,260 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import javaslang.Function2;
+import javaslang.Tuple;
+import javaslang.Tuple2;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.function.Function;
+
+import static javaslang.Function2.constant;
+import static javaslang.collection.BitMappedTrie.BRANCHING_BASE;
+import static javaslang.collection.BitMappedTrie.branchingFactor;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class VectorPropertyTest {
+    @Before
+    public void setUp() { BRANCHING_BASE = 2; }
+
+    @Test
+    public void shouldCreateAndGet() {
+        for (int i = 0; i < 2000; i++) {
+            final Seq<Integer> expected = Array.range(0, i);
+            final Vector<Integer> actual = Vector.ofAll(expected);
+            assertAreEqual(expected, actual);
+        }
+    }
+
+    @Test
+    public void shouldIterate() {
+        for (byte depth = 1; depth <= 6; depth++) {
+            for (int i = 0; i < getMaxSizeForDepth(depth); i++) {
```

Comment:
Since `getMaxSizeForDepth` currently has a cap of `10 000`, which requires depth = 3, you are iterating over the very same interval for depths `3 .. 6`.